### ISSUE TITLE
fyne get requires root access

### ISSUE THREAD

**Issue by paboum** [fyne-io/fyne#3425]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The `fyne get` command assumes `/usr/local/bin` as installation target (hardcoded) and makes no use of `GOPATH` env var.
```
$ fyne get fyne.io/apps
2022/11/24 05:08:10 Fyne error:  Failed to create dirrectory
2022/11/24 05:08:10   Cause: mkdir /usr/local/share: permission denied
2022/11/24 05:08:10   At: /usr/go/pkg/mod/fyne.io/fyne/v2@v2.2.4/cmd/fyne/internal/util/file.go:36
failed to copy application binary file: open /usr/local/bin/apps: permission denied
```

### How to reproduce

`$ fyne get fyne.io/apps`

### Example code

```go
	if _, err := os.Stat(filepath.Join("/", "usr", "local")); os.IsNotExist(err) {
		prefixDir = util.EnsureSubDir(util.EnsureSubDir(p.dir, tempDir), "usr")
		local = ""
	} else {
		prefixDir = util.EnsureSubDir(util.EnsureSubDir(util.EnsureSubDir(p.dir, tempDir), "usr"), "local")
	}

	shareDir := util.EnsureSubDir(prefixDir, "share")

	binDir := util.EnsureSubDir(prefixDir, "bin")
	binName := filepath.Join(binDir, filepath.Base(p.exe))
	err := util.CopyExeFile(p.exe, binName)
	if err != nil {
		return fmt.Errorf("failed to copy application binary file: %w", err)
	}

```

### Fyne version

2.2.4

### Go compiler version

1.19.2

### Operating system

Linux

### Operating system version

Gentoo

### Additional Information

It would be preferable to respect `GOPATH` and build apps in `$GOPATH/bin`, with a possibility of creating symlinks in `/usr/local/bin/`.
Also, note that `/usr/local/bin` may be writable for regular users on some systems, but it's not a standard behavior:
https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard

**Comment by andydotxyz** [comment]
The behaviour of "fyne get" is to install to the system specific application area - this are graphical app packages not go binaries. On windows this is Program Files and on macOS it is /Applications. This way the apps are findable in standard launchers.
Installing to the go binary location would not offer this benefit.
I agree that requiring sudo on some systems is undesirable, but I do not know what the alternative is?...

**Comment by andydotxyz** [comment]
Fyne install is not like go install - it is not simply dripping the binary file in. It has metadata and package artefacts depending on OS

**Comment by Jacalz** [comment]
How about adding a flag like `--local` that installs into `$HOME/.local`? We are already doing something similar with the installation Makefile for the packaged Linux/BSD applications: https://github.com/fyne-io/fyne/blob/2c1d0bd8a00beedf283923ad1d91d519b822b858/cmd/fyne/internal/templates/data/Makefile#L30.

**Comment by andydotxyz** [comment]
Maybe `--user` as that is a more generally understood term for user space installs?

**Comment by paboum** [comment]
> Maybe `--user` as that is a more generally understood term for user space installs?

How about best effort, if there is no write access to `/usr/local/bin`, then try `$HOME/.local/bin`? Unless, of course, `FYNEBIN` env var doesn't point to somewhere else.

**Comment by andydotxyz** [comment]
I don't think that it's a good idea to significantly change location based on permission setup.
Have you ever run a command and forgotten you need sudo? In this case it would potentially put the app in an unexpected place and exit with no error.

**Comment by Jacalz** [comment]
I agree. It is better to default to requiring sudo and then potentially give a suggestion, in the error message, to either elevate permissions (using `sudo`, `doas`, etc.) or use `--user` flag.

**Comment by paboum** [comment]
What information is still needed then?

**Comment by andydotxyz** [comment]
It would be good to know if you are happy with with @Jacalz proposed. If so we can proceed.

**Comment by Jacalz** [comment]
@paboum Why did you close this as not planned? I think it should stay open until the proposed feature has been implemented.

**Comment by NiceGuyIT** [comment]
I was attempting to install [fyneterm](https://github.com/fyne-io/terminal) when I ran across this issue. There are a few things that can be improved.
1. Document the need for sudo.
   - `fyne get --help` does not mention sudo.
   - The docs, <https://developer.fyne.io/search.html?q=sudo>, do not mention sudo other than installing the libraries.
2. `sudo ~/go/bin/fyne get github.com/fyne-io/terminal/cmd/fyneterm` builds and installs the application as root leaving behind `/root/go/{pkg/src}`.
3. Building as user and prompting for sudo/root access for installation is preferred.

**Comment by andydotxyz** [comment]
Prompting for `sudo` would be ideal, however it is not clear how we know whether `sudo` or `su` is the setup for a local system - neither is universal. If you have hints and/or interest in creating a PR that would be excellent!

**Comment by Bluebugs** [comment]
Maybe another approach is to detect the user we are running under and if it isn't root, we do the install as a user (our Unix Makefile support that with the `user-install` rule). That might be easier.

**Comment by Jacalz** [comment]
We have already discussed the solutions for this further up in this thread. If believe we already have a good idea for how to solve this

**Comment by Bluebugs** [comment]
> We have already discussed the solutions for this further up in this thread. I believe we already have a good idea for how to solve this

Oops, indeed.

**Comment by andydotxyz** [comment]
Moving to the tools repo where this development is being picked up.
